sphinx/search/__init__.py

```diff
@@ -503,7 +503,9 @@ def _visit_nodes(node):
                 word_store.words.extend(split(node.astext()))
             elif isinstance(node, nodes.title):
                 title = node.astext()
-                ids = node.parent['ids']
+                ids = node.parent['ids'] or node.parent.parent['ids']
+                if not ids:
+                    return
```

Comment:
Note: there is also a (legacy?) equivalent [code block in `WordCollector`](https://github.com/sphinx-doc/sphinx/blob/e976059fd6ab49b1c2784445d7095c9287897724/sphinx/search/__init__.py#L232-L236).